### ISSUE TITLE
Remove or rename fromJSON to fromArray

### ISSUE THREAD

**Issue by quiquelhappy** [purecoreio/purecore-js#3]
fromJSON is not technically correct, as the value being passed is not JSON, but a javascript array, as it is always converted from JSON with JSON.parse() first. fromJSON should be listening to STRINGS ONLY as thats the only thing officially considered JSON in javascript.

Please, rename fromJSON to fromArray (correct term) or refractor fromArray to look like fromJSON and remove fromJSON definetely (fromJSON can look like fromJSON(json:string){return object,fromArray(JSON.parse(json))}

**Comment by quiquelhappy** [comment]
@networkException

**Comment by networkException** [comment]
I do see how JavaScriptObjectNotation should only apply to Strings, as such I will rework the fromJSON(json: any) to fromJSON(json: string) methods. 

However there are two main issues with adding a fromArray(array: any) method:

- fromArray already does exist, but it is deprecated due to the fact that it takes an existing object and changes the fields. fromJSON in contrast currently uses the objects constructor to create a new instance of it, resulting in a more [pure functional](https://en.m.wikipedia.org/wiki/Pure_function) behavior.
- As you bring up JavaScript terminology for JSON, it should be noted that an Array in JavaScript looks like this [], it's only properties being values at an index (for example array[1] = "a"). Therefore I propose for a behavior like applying an already existing json object to instantiate a new object from it the name "fromObject".

In conclusion a class in the wrapper would have 3 methods: 
- fromArray(array: any) deprecated, only existing for backwards compatibility
- fromJSON(json: string) parsing the json and calling fromObject with the result
- fromObject(object: Object) using the object's properties to create a new instance of the class it's being called on

Network.fromJSON('{"uuid": "xyz"}'); 
Network.fromObject({ uuid: "xyz" });

**Comment by networkException** [comment]
I would be happy to completely remove fromArray as I stated it in the comment above, however I do not know if you want to break the current use of it in depending projects
